### PR TITLE
improving builder pattern for Event

### PR DESCRIPTION
**Pull request by pwittchen**

### CODE REVIEW

**Comment by davidmoten** [comment]
Nearly there: 
* make Event final
* Event fields should be final
* Builder no-arg constructor should be present but private
* add 3 static entry methods so that people don't have to call `Event.builder().id()...` but can call `Event.id()...":
```java
public static Builder id(String id) {
    return builder().id(id);
}

public static Builder name(String name) {
    return builder().name(name);
}

public static Builder data(Serializable data) {
    return builder().data(data);
}
```
* add `Event.create()`:
```java
public static create() {
    return Event.builder().build();
}
```
* make Builder final
* up to you but at this point you don't need the `Event.builder()` method

**Comment by pwittchen** [comment]
Ok, I think I got it now :). I've pushed next commit with further improvements on this branch.

**Comment by pwittchen** [comment]
I squashed all my commits into one to make changes more clear.

File: library/src/main/java/com/github/pwittchen/reactivebus/library/Event.java [hunk 1]
```diff
@@ -73,27 +93,27 @@ public int hashCode() {
...
     }
 
-    public Builder setData(Serializable data) {
+    public Builder data(Serializable data) {
       this.data = data;
       return this;
     }
 
-    public Event build() {
+    public Event create() {
```
Comment (davidmoten) [inline library/src/main/java/com/github/pwittchen/reactivebus/library/Event.java]: This should still be called `build()` (you *build* with a builder).

File: library/src/main/java/com/github/pwittchen/reactivebus/library/Event.java [hunk 2]
```diff
@@ -73,27 +93,27 @@ public int hashCode() {
     return Objects.hash(id, name);
   }
 
-  public static class Builder {
+  public final static class Builder {
     private String id = UUID.randomUUID().toString();
     private String name = "";
     private Serializable data;
```
Comment (davidmoten) [inline library/src/main/java/com/github/pwittchen/reactivebus/library/Event.java:100]: Prevent instantiation of Builder by adding private do-nothing constructor:
```java
private Builder() {
}
```

**Review by davidmoten — APPROVED** [review]

File: library/src/main/java/com/github/pwittchen/reactivebus/library/Event.java (hunk 1 above)
Comment (pwittchen) [reply library/src/main/java/com/github/pwittchen/reactivebus/library/Event.java]: Fixed.

File: library/src/main/java/com/github/pwittchen/reactivebus/library/Event.java (hunk 2 above)
Comment (pwittchen) [reply library/src/main/java/com/github/pwittchen/reactivebus/library/Event.java:100]: Fixed.

**Comment by pwittchen** [comment]
Ok, I think it looks fine now and I can merge it to the master branch. Thanks a lot for the code review @davidmoten!

**Comment by davidmoten** [comment]
Nice. Welcome to the world of nice-to-use builders!

**Pull request merged**